core/trino-spi/src/main/java/io/trino/spi/connector/ColumnMetadata.java

```diff
@@ -38,7 +40,8 @@
     private final boolean hidden;
     private final Map<String, Object> properties;
 
-    public ColumnMetadata(String name, Type type)
+    @JsonCreator
+    public ColumnMetadata(@JsonProperty("name") String name, @JsonProperty("type") Type type)
```

Comment:
I'm not sure if this is the right constructor to use for serialization / deserialization purposes. Some of the fields (e.g.: comment, extraInfo, properties) might get lost during serialization roundtrip

Reply:
Yes, I answer above.